code/code/sys/indexdata.cc

```diff
@@ -18,167 +18,108 @@ extern FILE *mob_f;
 std::vector<mobIndexData> mob_index;
 std::vector<objIndexData> obj_index;
 
-indexData::indexData() :
-  virt(0),
-  pos(0),
-  number(0),
-  name(NULL),
-  short_desc(NULL),
-  long_desc(NULL),
-  description(NULL),
-  max_exist(-99),
-  spec(0),
-  weight(0)
-{
-}
-
-indexData & indexData::operator= (const indexData &a)
-{
-  if (this == &a)
-    return *this;
-
-  virt = a.virt;
-  pos = a.pos;
-  number = a.number;
-  delete [] name;
-  delete [] short_desc;
-  delete [] long_desc;
-  delete [] description;
-  name = mud_str_dup(a.name);
-  short_desc = mud_str_dup(a.short_desc);
-  long_desc = mud_str_dup(a.long_desc);
-  description = mud_str_dup(a.description);
-
-  max_exist = a.max_exist;
-  spec = a.spec;
-  weight = a.weight;
-
-  return *this;
-}
+indexData::indexData()
+    : virt(0),
+      pos(0),
+      number(0),
+      name(nullptr),
+      short_desc(nullptr),
+      long_desc(nullptr),
+      description(nullptr),
+      max_exist(-99),
+      spec(0),
+      weight(0) {}
 
 indexData::indexData(const indexData &a) :
   virt(a.virt),
   pos(a.pos),
   number(a.number),
+  name(a.name),
+  short_desc(a.short_desc),
+  long_desc(a.long_desc),
+  description(a.description),
   max_exist(a.max_exist),
   spec(a.spec),
-  weight(a.weight)
-{
-  name = mud_str_dup(a.name);
-  short_desc = mud_str_dup(a.short_desc);
-  long_desc = mud_str_dup(a.long_desc);
-  description = mud_str_dup(a.description);
-}
-
-indexData::~indexData()
-{
-  delete [] name;
-  delete [] short_desc;
-  delete [] long_desc;
-  delete [] description;
-}
-
-mobIndexData::mobIndexData() :
-  faction(-99),
-  Class(-99),
-  level(-99),
-  race(-99),
-  doesLoad(false),
-  numberLoad(0)
-{
-}
-
-mobIndexData & mobIndexData::operator= (const mobIndexData &a)
-{
-  if (this == &a)
-    return *this;
-
-  indexData::operator=(a);
-
-  faction = a.faction;
-  Class = a.Class;
-  level = a.level;
-  race = a.race;
-  doesLoad = a.doesLoad;
-  numberLoad = a.numberLoad;
-
-  return *this;
-}
-
-mobIndexData::mobIndexData(const mobIndexData &a) :
-  indexData(a),
-  faction(a.faction),
-  Class(a.Class),
-  level(a.level),
-  race(a.race),
-  doesLoad(a.doesLoad),
-  numberLoad(a.numberLoad)
-{
-}
-
-mobIndexData::~mobIndexData()
-{
+  weight(a.weight) {}
+
+mobIndexData::mobIndexData()
+    : faction(-99),
+      Class(-99),
+      level(-99),
+      race(-99),
+      doesLoad(false),
+      numberLoad(0) {}
+
+mobIndexData::mobIndexData(int _virt, const sstring &_name, const sstring &_short_desc,
+                           const sstring &_long_desc, const sstring &_description, int _max_exist,
+                           int _spec, float _weight, long faction, long Class, long level,
+                           long race)
+    : faction(faction), Class(Class), level(level), race(race) {
```

Comment:
Why are you using different styles of initialization for different variables?

Reply:
Addressed in https://github.com/sneezymud/sneezymud/pull/380/commits/ca9f8b35b90fb1f6818a43d8dc48feadd8831a56